airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py

```diff
@@ -0,0 +1,349 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qsl, urlparse
+
+import pendulum
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator
+
+# See the get_class_ids function explanation
+auxiliary_id_map = {}
+
+
+class SnapchatMarketingException(Exception):
+    """ Just for formatting the exception as SnapchatMarketing"""
+
+
+class SnapchatMarketingStream(HttpStream, ABC):
+    # The default value that is returned by stream_slices if there is no slice found: [None]
+    default_stream_slices_return_value = [None]
+    url_base = "https://adsapi.snapchat.com/v1/"
+    primary_key = "id"
+
+    @property
+    def response_root_name(self):
+        return self.__class__.__name__.lower()
+
+    @property
+    def response_item_name(self):
+        return self.response_root_name[:-1]
+
+    def __init__(self, start_date, **kwargs):
+        super().__init__(**kwargs)
+        self.start_date = pendulum.parse(start_date).to_rfc3339_string()
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page_cursor = response.json().get("paging", False)
+        if next_page_cursor:
+            return {"cursor": dict(parse_qsl(urlparse(next_page_cursor["next_link"]).query))["cursor"]}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return next_page_token if next_page_token else {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        json_response = response.json().get(self.response_root_name)
+        for resp in json_response:
+            yield resp.get(self.response_item_name)
+
+
+class IncrementalSnapchatMarketingStream(SnapchatMarketingStream, ABC):
+    cursor_field = "updated_at"
+    superior_cls = None
```

Comment:
superior_cls  - I would change this name to something more 'stream' related name, like depends_on_stream, main_stream, or something like this

Reply:
Good idea. Change it to `depends_on_stream`